vars/skipStage.groovy

```diff
@@ -22,6 +22,15 @@ boolean skip_ftest(String distro) {
            skip_stage_pragma('func-test-' + distro)
 }
 
+boolean skip_ftest_valgrind(String distro) {
+    return distro == 'ubuntu20' ||
+           skip_stage_pragma('func-test') ||
+           skip_stage_pragma('func-test-vm') ||
+           skip_stage_pragma('func-test-vm-valgrind') ||
+           ! testsInStage() ||
+           skip_stage_pragma('func-test-' + distro)
```

Comment:
I know you have not requested reviews yet, so you still have the right to force-push to this PR if you want.  Just thought I'd make a suggestion though.  You can call the function you copied instead of copying all of it's conditions (and creating a future situation where one gets updated with a new condition and the other doesn't):

```suggestion
    return skip_ftest(distro) ||
           skip_stage_pragma('func-test-vm-valgrind')
```

Reply:
Please leave comments for the comment creator to resolve, which they should/could do once they have verified that the resolution applied satisfies their concern.